### PR TITLE
fix: mv id() to getObjectID().index because id() returns ObjectID, not uint_t, in podio v0.17.1

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This adds forward compatibility with podio v0.17.1. See https://github.com/AIDASoft/podio/pull/493

### What kind of change does this PR introduce?
- [x] Bug fix (issue: compilation falures with podio 0.17.1)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No. The modified algorithms are not relying on the most significant bits in the previous ObjectID uint_t that encode the collectionID.

### CODE REVIEW

File: src/algorithms/pid/MergeParticleID.cc [hunk 1]
```diff
@@ -94,7 +94,7 @@ std::unique_ptr<edm4eic::CherenkovParticleIDCollection> eicrecon::MergeParticleI
         m_log->error("PID object found with no charged particle");
         continue;
       }
-      auto id_particle = charged_particle_track_segment.id();
+      auto id_particle = charged_particle_track_segment.getObjectId().index;
```
Comment (wdconinc) [inline src/algorithms/pid/MergeParticleID.cc:97]:
```suggestion
      auto id_particle = charged_particle_track_segment.getObjectID().index;
```

**Review by veprbl — APPROVED** [review]
For correctness sake, we should be leaning towards full object id comparisons where possible. This looks correct, still.

**Comment by wdconinc** [comment]
> For correctness sake, we should be leaning towards full object id comparisons where possible.

Fortunately that is already the case in several other places. I didn't (need to) replace those cases because the ObjectID comparison works fine. It's only the places where the ObjectID is explicitly assumed to be a uint_t (for storing or debug output) that are modified here.

**Comment by epic-capybara** [comment]
Successfully created backport PR for `v1.7`:
- #1109

**Pull request merged**